validator/patch.go

```diff
@@ -14,10 +14,16 @@ import (
 
 // GetPatchedProject creates and validates a project created by fetching latest commit information from GitHub
 // and applying the patch to the latest remote configuration. The error returned can be a validation error.
-func GetPatchedProject(ctx context.Context, p *patch.Patch, githubOauthToken string) (*model.Project, error) {
+func GetPatchedProject(ctx context.Context, p *patch.Patch, githubOauthToken string) ([]byte, error) {
```

Comment:
Split this function in two: 
1. apply the patch
2. validate the config file

This way if something goes wrong applying the patch we don't tell the user their configuration file is invalid

Reply:
I think the function names are a bit unintuitive with regards to expected use. Can we load the project in this function and then just call validate in the next?

It's also super weird that this is in the validator project, now that I think about it.

Reply:
I was thinking of errors unmarshalling and translating as part of the validation.

Reply:
@tychoish: Which package would you suggest putting them in?

Reply:
Now loads the project in GetPatchedProject
Moved to a more sensible place (thirdparty was a import cycle so I put it in patch_lifecycle next to `MakePatchedConfig`)